test/test_kms_with_cli.py

```diff
@@ -0,0 +1,197 @@
+
+    print('====================test_Stest_RSA3072_sign_verify end===========================')
+
+def test_GenerateDataKeyWithoutPlaintext(base_url, headers):
+    print('====================test_GenerateDataKeyWithoutPlaintext start===========================')
+    key = createkey.createkey(base_url, "EH_AES_GCM_128", "EH_INTERNAL_KEY")
+
+    # test GenerateDataKeyWithoutPlaintext
+    aad = str(base64.b64encode("test".encode("utf-8")),'utf-8')
+
+    ciphertext = generate_datakey_withoutplaint.generate_datakey_withoutplaint(base_url, key, 48, aad)
+
+    # test Decrypt(cipher_datakey)
+    decrypt.decrypt(base_url, key, ciphertext, aad)
+
+    print('====================test_GenerateDataKeyWithoutPlaintext end===========================')
+
+def test_GenerateDataKey(base_url, headers):
+    print('====================test_GenerateDataKey start===========================')
+    key = createkey.createkey(base_url, "EH_AES_GCM_128", "EH_INTERNAL_KEY")
+    
+    # test GenerateDataKey
+    aad = str(base64.b64encode("test".encode("utf-8")),'utf-8')
+    ciphertext = generate_datakey.generate_datakey(base_url, key, 16, aad)
+
+    # test Decrypt(cipher_datakey)
+    decrypt.decrypt(base_url, key, ciphertext, aad)
+
+    print('====================test_GenerateDataKey end===========================')
+
+def test_AES128(base_url, headers):
+    print('====================test_AES128 start===========================')
+    key = createkey.createkey(base_url, "EH_AES_GCM_128", "EH_INTERNAL_KEY")
+
+    # test Encrypt("123456")
+    data = str(base64.b64encode("123456".encode("utf-8")),'utf-8')
+    aad = str(base64.b64encode("test".encode("utf-8")),'utf-8')
+    ciphertext = encrypt.encrypt(base_url, key, data, aad)
+
+    # test Decrypt(ciphertext)
+    plaintext = decrypt.decrypt(base_url, key, ciphertext, aad)
+    
+    print('Decrypt plaintext:\n%s\n' %(plaintext))
+    print('check Decrypt plaintext result with %s: %s\n' %('123456', plaintext == '123456'))
+
+    print('====================test_AES128 end===========================')
+
+# A hook function forst JSON load function that avoid
+# bool type value changing from "true" to "True".It will be caused the sign verfiy error
+def no_bool_convert(pairs):
+  return {k: str(v).casefold() if isinstance(v, bool) else v for k, v in pairs}
+
+def test_GenerateQuote_and_VerifyQuote(base_url, headers):
+    print('====================test_GenerateQuote_and_VerifyQuote start===========================')
+    generate_quote.generate_quote_with_file(base_url, "a.txt")
+
+    verify_quote.verify_quote_with_file(base_url, "a.txt")
+
+    print('====================test_GenerateQuote_and_VerifyQuote end===========================')
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--ip-adress', type=str, help='ip address of the ehsm_kms_server', required=True)
+    parser.add_argument('-p', '--port', type=str, help='port of the ehsm_kms_server', required=True)
```

Comment:
please refine it, and align it same with the cli scripts, e.g
test.py --url http://1.2.3.4:9000/

Reply:
fixed